hazelcast/src/main/java/com/hazelcast/concurrent/lock/LockServiceImpl.java

```diff
@@ -338,11 +342,21 @@ private void clearLockStoresHavingLesserBackupCountThan(int partitionId, int thr
 
     @Override
     public DistributedObject createDistributedObject(String objectId) {
-        return new LockProxy(nodeEngine, this, objectId);
+        LockProxy lockProxy = new LockProxy(nodeEngine, this, objectId);
+
+        if (logger.isFinestEnabled()) {
+            logger.finest("Created proxy for lock " + objectId);
+        }
+
+        return lockProxy;
     }
 
     @Override
     public void destroyDistributedObject(String objectId) {
+        if (logger.isFinestEnabled()) {
```

Comment:
I would prefer removing the proxy creation logging. Proxies are irrelevant. The most important part is what is happening on the partition owner.